protocol/daemons/pricefeed/client/price_fetcher/price_fetcher.go

```diff
@@ -269,8 +303,16 @@ func (pf *PriceFetcher) runSubTask(
 			price.LastUpdatedAt,
 		)
 
+		// Report market as available.
+		availableMarkets[price.MarketId] = true
+
 		pf.writeToBufferedChannel(exchangeId, price, err)
 	}
+
+	// Emit metrics on this exchange's market availability.
+	for marketId, available := range availableMarkets {
```

Comment:
For my understanding what is the cadence of this logic? Is it per `runSubTask` run? How often is it?

Reply:
It's run once every time a price fetcher makes a query.